cubeviz/tools/collapse_cube.py

```diff
@@ -311,187 +228,321 @@ def hide_start_end(self, dohide):
+            self.ui.error_label.setText('Start index must be less than the end index.')
+            self.ui.error_label.setVisible(True)
+            return None, None
+
+        log.debug('  returning with start_index {} and end_index {}'.format(
+            start_index, end_index))
+
+        return start_index, end_index
+
+    def _calculate_callback_simple_sigma_check(self):
+        log.debug('_calculate_callback_simgple_sigma_check')
+
+        simple_sigma = self.ui.simple_sigma_input.text().strip()
+        log.debug('    simple_sigma {}'.format(simple_sigma))
+
+        try:
+            simple_sigma = float(simple_sigma)
+        except:
+            self.ui.simple_sigma_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Sigma must be a floating point number.')
+            self.ui.error_label.setVisible(True)
+            return None
+
+        return simple_sigma
+
+    def _calculate_callback_advanced_sigma_check(self):
+        sigma = self.ui.advanced_sigma_input.text().strip()
+        sigma_lower = self.ui.advanced_sigma_lower_input.text().strip()
+        sigma_upper = self.ui.advanced_sigma_upper_input.text().strip()
+        sigma_iters = self.ui.advanced_sigma_iters_input.text().strip()
+
+        if len(sigma) > 0:
+            try:
+                sigma = float(sigma)
+            except:
+                self.ui.advanced_sigma_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+                self.ui.error_label.setText('Sigma must be a floating point number.')
+                self.ui.error_label.setVisible(True)
+                return None
         else:
-            self.start_label.show()
-            self.start_text.show()
-            self.end_label.show()
-            self.end_text.show()
+            self.ui.advanced_sigma_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Sigma must be a floating point number and not empty.')
+            self.ui.error_label.setVisible(True)
+            return None, None, None, None
+
+        if len(sigma_lower) > 0:
+            try:
+                sigma_lower = float(sigma_lower)
+            except:
+                self.ui.advanced_sigma_lower_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+                self.ui.error_label.setText('Lower sigma must be a floating point number.')
+                self.ui.error_label.setVisible(True)
+                return None*4
+        else:
+            sigma_lower = None
+
+        if len(sigma_upper) > 0:
+            try:
+                sigma_upper = float(sigma_upper)
+            except:
+                self.ui.advanced_sigma_upper_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+                self.ui.error_label.setText('Upper sigma must be a floating point number.')
+                self.ui.error_label.setVisible(True)
+                return None*4
+        else:
+            sigma_upper = None
+
```

Comment:
There's no check to make sure `sigma_lower < sigma_upper`, or to make sure that either of these values are `<= 0`.

Reply:
I don't think a check for <= 0 is worth doing here. Someone could easily subtract sky and end up wanting ```sigma_lower <= 0.``` It would be unusual to want ```sigma_upper <= 0.``` 

It's good to check that sigma_lower < sigma_upper...but I wonder if ```astropy.stats.sigma_clip``` does that, in which case trapping the error might be a better approach than doing the check before passing the parameters.